src/util/package-name-utils.js

```diff
@@ -0,0 +1,13 @@
+/* @flow */
+
+export function getPlatformSpecificPackageFilename(pkg: {name: string, version: string}): string {
+  // TODO support hash for all subdependencies that have installs scripts
```

Comment:
This will be problematic, yeah :/ Something I had in mind was that maybe we could implement nested lockfiles to solve this issue. Given a nested lockfile, we could guarantee that a package would always be built against a specific set of dependencies, and we could cache it more efficiently. Wdyt?

Reply:
I think nested lockfile is an overkill and may be confusing considering that we don't support nested lockfiles in general.
We might generate something in between - a lock-like file that will only have `dependencies` section because we don't really need "version" and "resolved" fields as they are already defined in the parent yarn.lock

Reply:
On the other hand having that file inside the .tgz file requires us to unzip it before checking whether the archive works for us, we need to encode all dependencies into the file suffix and hashing them seems a reasonable way to shorten the suffix

Reply:
Anyway, we'll discuss it in V2 of this feature, we need to see how it will be used by the users if it will be :)